molgenis-data-postgresql/src/main/java/org/molgenis/data/postgresql/PostgreSqlQueryGenerator.java

```diff
@@ -494,9 +494,33 @@ private static String getSqlDropTable(String tableName)
 					result.append(" NOT ");
 					break;
 				case RANGE:
-					// TODO implement RANGE query operator
-					throw new UnsupportedOperationException(format(
-							"Query operator [%s] not supported by PostgreSQL repository", r.getOperator().toString()));
+					Object range = r.getValue();
+					if (range == null)
+					{
+						throw new MolgenisDataException("Missing value for RANGE query");
+					}
+					if (!(range instanceof Iterable<?>))
+					{
+						throw new MolgenisDataException(format("RANGE value is of type [%s] instead of [Iterable]",
+								range.getClass().getSimpleName()));
+					}
+					Iterator<Object> rangeValues = ((Iterable) range).iterator();
+					parameters.add(rangeValues.next()); // smaller
```

Comment:
I used the variable names from the ```rng``` method of ```QueryImpl``` to stay consistent. I agree that from and to are better :)